tutorials/tts/4_TTS_FastPitch_Finetuning.ipynb

```diff
@@ -0,0 +1,528 @@
+    "    \"\"\"\n",
+    "    if not mixing_enabled:\n",
+    "        exp_dir = \"{}/{}_to_{}_no_mixing_{}_mins\".format(experiment_base_dir, original_speaker_id, new_speaker_id, duration_mins)\n",
+    "    else:\n",
+    "        exp_dir = \"{}/{}_to_{}_mixing_{}_mins\".format(experiment_base_dir, original_speaker_id, new_speaker_id, duration_mins)\n",
+    "    \n",
+    "    ckpt_candidates = []\n",
+    "    last_ckpt = None\n",
+    "    for root, dirs, files in os.walk(exp_dir):\n",
+    "        for file in files:\n",
+    "            if file.endswith(\".ckpt\"):\n",
+    "                val_error = float(file.split(\"v_loss=\")[1].split(\"-epoch\")[0])\n",
+    "                if \"last\" in file:\n",
+    "                    last_ckpt = os.path.join(root, file)\n",
+    "                ckpt_candidates.append( (val_error, os.path.join(root, file)))\n",
+    "    ckpt_candidates.sort()\n",
+    "    \n",
+    "    return ckpt_candidates, last_ckpt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0153bd5a",
+   "metadata": {},
+   "source": [
+    "Specify the speaker id, duration mins and mixing variable to find the relevant checkpoint from the exp_base_dir and compare the synthesized audio with validation samples of the new speaker."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8901f88b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "new_speaker_id = 92\n",
+    "duration_mins = 5\n",
+    "mixing = False\n",
+    "original_speaker_id = 8051\n",
+    "\n",
+    "\n",
+    "_ ,last_ckpt = get_best_ckpt(exp_base_dir, new_speaker_id, duration_mins, mixing, original_speaker_id)\n",
+    "print(last_ckpt)\n",
+    "\n",
+    "spec_model = FastPitchModel.load_from_checkpoint(last_ckpt)\n",
+    "spec_model.eval().cuda()\n",
+    "_speaker=None\n",
+    "if mixing:\n",
+    "    _speaker = 1\n",
+    "\n",
+    "num_val = 2\n",
+    "\n",
+    "manifest_path = os.path.join(filelist_dir, \"{}_mainifest_dev_ns_all_local.json\".format(new_speaker_id))\n",
+    "val_records = []\n",
+    "with open(manifest_path, \"r\") as f:\n",
+    "    for i, line in enumerate(f):\n",
+    "        val_records.append( json.loads(line) )\n",
+    "        if len(val_records) >= num_val:\n",
+    "            break\n",
+    "            \n",
+    "for val_record in val_records:\n",
+    "    print (\"Real validation audio\")\n",
+    "    ipd.display(ipd.Audio(val_record['audio_filepath'], rate=44100))\n",
+    "    print (\"SYNTHESIZED FOR -- Speaker: {} | Dataset size: {} mins | Mixing:{} | Text: {}\".format(new_speaker_id, duration_mins, mixing, val_record['text']))\n",
+    "    spec, audio = infer(spec_model, vocoder, val_record['text'], speaker = _speaker)\n",
+    "    ipd.display(ipd.Audio(audio, rate=44100))\n",
+    "    %matplotlib inline\n",
+    "    #if spec is not None:\n",
+    "    imshow(spec, origin=\"lower\", aspect = \"auto\")\n",
+    "    plt.show()"
+   ]
+  },
+  {
```

Comment:
I think it would also be cool to show how people can fine tune their model on other data. Maybe even use their own voice after recording a few samples.

Reply:
Can move this to another PR